source/common/stats/symbol_table_impl.h

```diff
@@ -73,6 +73,8 @@ class SymbolTableImpl : public SymbolTable {
    */
   class Encoding {
   public:
+    class TokenIter;
```

Comment:
Do you seeing this being used outside the SymbolTable? If not, could this be private and can SymbolTable be a friend?